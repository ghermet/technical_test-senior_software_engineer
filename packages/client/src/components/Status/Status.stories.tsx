import React from 'react'
import Status from './Status'
import '../../styles/globals.generated.css'

export default {
  title: 'Components/Status',
  component: Status,
}

export const Protected = () => <Status status="PROTECTED" />
export const Vulnerable = () => <Status status="VULNERABLE" />
