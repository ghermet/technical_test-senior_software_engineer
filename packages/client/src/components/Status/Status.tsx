import React from 'react'
import { ShieldCheck, ExclamationMarkTriangle } from '@pretto/picto'
import { ctwMerge } from '../../utility/ctw-merge'
import type { PeopleStatus as SearchPeopleStatus } from '../../graphql/queries/PeopleByPage'

export const Status = ({ status }: { status: SearchPeopleStatus }) => {
  const isProtected = status === 'PROTECTED'
  const isVulnerable = status === 'VULNERABLE'
  const statusText = isProtected ? 'PROTECTED' : 'Vulnerable'
  const statusIcon = isProtected ? (
    <ShieldCheck className={'fill-primary2'} />
  ) : (
    <ExclamationMarkTriangle className={'fill-error1'} />
  )
  return (
    <span
      className={ctwMerge('flex items-center flex-nowrap gap-2 body-book-16 capitalize', {
        'text-primary2': isProtected,
        'text-error1': isVulnerable,
      })}
    >
      {statusIcon} {statusText.toLowerCase()}
    </span>
  )
}

export default Status
