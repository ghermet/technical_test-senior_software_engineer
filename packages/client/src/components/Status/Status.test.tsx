import { render } from '@testing-library/react'
import React from 'react'
import Status from './Status'

describe('Status', () => {
  it('should render a PROTECTED status', () => {
    const { container } = render(<Status status="PROTECTED" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a VULNERABLE status', () => {
    const { container } = render(<Status status="VULNERABLE" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
