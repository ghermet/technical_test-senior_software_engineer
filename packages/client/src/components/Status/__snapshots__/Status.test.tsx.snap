// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should render a PROTECTED status 1`] = `
<span
  class="flex items-center flex-nowrap gap-2 body-book-16 capitalize text-primary2"
>
  <svg
    class="fill-primary2"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12,0C8.63,2.87,6.52,3,3,3V14.53c0,4.61,3.2,5.81,9,9.47,5.8-3.66,9-4.86,9-9.47V3C17.48,3,15.37,2.87,12,0Zm0,1.29A11.5,11.5,0,0,0,20,4V14.53c0,3.9-2.64,5-8,8.29-5.38-3.35-8-4.41-8-8.29V4A11.5,11.5,0,0,0,12,1.29ZM17,9l-6,6.62L7.34,11.72,8,11l3,3.18,5.22-5.87Z"
    />
  </svg>
   
  protected
</span>
`;

exports[`Status should render a VULNERABLE status 1`] = `
<span
  class="flex items-center flex-nowrap gap-2 body-book-16 capitalize text-error1"
>
  <svg
    class="fill-error1"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M24,23H0L12,1ZM1.68,22H22.32L12,3.09ZM12,20a.85.85,0,1,0-.85-.84A.85.85,0,0,0,12,20Zm.5-11v8h-1V9Z"
    />
  </svg>
   
  vulnerable
</span>
`;
